### ISSUE TITLE
Websocket message callback is not called on connect

### ISSUE THREAD

**Issue by egelja** [curl/curl#11402]
### I did this

I started with the [`websocket-cb` example](https://github.com/curl/curl/blob/master/docs/examples/websocket-cb.c) and connected to an echo websocket server for testing.

### I expected the following

I expected the data callback to be called once the connection is established to allow sending the first message in the connection. However, it isn't, and I cannot send a message without receiving one first.

### curl/libcurl version

```
libcurl/8.1.2 Schannel zlib/1.2.13
```

### operating system

```
Windows_NT ArthurDent 10.0 22621 x86_64 MS/Windows (Windows 11)
```

**Comment by dfandrich** [comment]
Is this the issue you're reporting? https://curl.se/mail/lib-2023-06/0006.html

**Comment by egelja** [comment]
Yes, it is. Thank you, I tried to search the mailing lists but could not find it.

**Comment by bagder** [comment]
There is currently no attempt of doing this and no documentation suggests libcurl would do this, so I consider this a (fair) feature-request.

**Comment by bagder** [comment]
@egelja any chance you're interested in working on adding this feature?

**Comment by egelja** [comment]
> @egelja any chance you're interested in working on adding this feature?

Possibly, although I would not be able to right now. 

Is there an API you had in mind for this enhancement?

**Comment by bagder** [comment]
The API needs to be worked out.

**Comment by bagder** [comment]
This is a feature-request and not a bug so we add this to the TODO for someone to implement.

**Comment by polarina** [comment]
I was able to achieve the same by waiting for `\r\n` to be written to the CURLOPT_HEADERFUNCTION callback, after which `curl_ws_send` works fine.

**Comment by chillinbythetree** [comment]
Is this fixed?
